From 1827f9ee18d30372105802f428888fe83ad190ba Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Thu, 28 Jul 2022 14:11:07 +0900
Subject: [PATCH] add virtual dtor for base::SharedMemoryMapper

update_engine has -Wnon-virtual-dtor buildflag which causes compile
error.

Change-Id: I97eb841037a52febf4e495b6eeeec37f01c54d23
---
 base/memory/shared_memory_mapper.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/base/memory/shared_memory_mapper.h b/base/memory/shared_memory_mapper.h
index 3cdff54ff6da..4e3eacc5bc3f 100644
--- a/base/memory/shared_memory_mapper.h
+++ b/base/memory/shared_memory_mapper.h
@@ -37,6 +37,8 @@ class BASE_EXPORT SharedMemoryMapper {
   // Unmaps the specified region of shared memory from the caller's address
   // space.
   virtual void Unmap(span<uint8_t> mapping) = 0;
+
+  virtual ~SharedMemoryMapper() = default;
 };
 
 }  // namespace base
-- 
2.37.1.359.gd136c6c3e2-goog

